### PR TITLE
GPG-655: Add Filter for anonymised users

### PR DESCRIPTION
**Pull request by xonde**

### CODE REVIEW

File: GenderPayGap.WebUI/Services/EmailSendingServiceHelpers.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ public static void SendUserAddedEmailToExistingUsers(Organisation organisation,
         {
             IEnumerable<string> emailAddressesForOrganisation = organisation.UserOrganisations
                 .Select(uo => uo.User.EmailAddress)
-                .Where(ea => ea != addedUser.EmailAddress);
+                .Where(ea => ea != addedUser.EmailAddress && ea != "anonymised");
```
Comment (malinaslevoaca1) [inline GenderPayGap.WebUI/Services/EmailSendingServiceHelpers.cs:14]: We should use `HasBeenAnonymised` instead.

**Review by malinaslevoaca1 — CHANGES_REQUESTED** [review]
I know most (if not all) of the failures come from new user registrations so it's good to have this filter but I think we should also check if the email address is anonymised before sending any email (as in [here](https://github.com/cabinetoffice/gender-pay-gap/blob/d293c0d35f2a7253901bcb91f62be536b8f816bf/GenderPayGap.WebUI/Services/EmailSendingService.cs#L424))

**Comment by xonde** [comment]
> I know most (if not all) of the failures come from new user registrations so it's good to have this filter but I think we should also check if the email address is anonymised before sending any email (as in [here](https://github.com/cabinetoffice/gender-pay-gap/blob/d293c0d35f2a7253901bcb91f62be536b8f816bf/GenderPayGap.WebUI/Services/EmailSendingService.cs#L424))

Ah yes, I initially made the change here but I wasn't sure if that was what ticket wanted, will revert it to that now

**Comment by malinaslevoaca1** [comment]
> > I know most (if not all) of the failures come from new user registrations so it's good to have this filter but I think we should also check if the email address is anonymised before sending any email (as in [here](https://github.com/cabinetoffice/gender-pay-gap/blob/d293c0d35f2a7253901bcb91f62be536b8f816bf/GenderPayGap.WebUI/Services/EmailSendingService.cs#L424))
> 
> Ah yes, I initially made the change here but I wasn't sure if that was what ticket wanted, will revert it to that now

@xonde Well, we could keep this filter but it should be there as well

**Review by malinaslevoaca1 — APPROVED** [review]
Nice

**Pull request merged**